packages/enzyme-test-suite/test/Adapter-spec.jsx

```diff
@@ -801,4 +801,69 @@ describe('Adapter', () => {
       },
     }));
   });
+
+  it.only('should determine valid element types', () => {
+    const renderer = adapter.createRenderer({ mode: 'shallow' });
+
+    class Component extends React.Component {
+      render() { return null }
+    }
+    const ArrowFunction = () => null
+
+    expect(adapter.isValidElementType('div')).to.equal(true);
+    expect(adapter.isValidElementType(Component)).to.equal(true);
+    expect(adapter.isValidElementType(ArrowFunction )).to.equal(true);
+
+    if (createPortal) {
```

Comment:
instead of using if checks, these should be checking the React versions and running conditionslly, as their own it’s